### PR TITLE
[STORM-1144] Provide resource (mem/cpu) assign feedback for schedulers in UI

### PR DESCRIPTION
**Pull request by zhuoliu**
1. Calculate the requested and assigned onheap memory, offheap memory and CPU.
2. Add data structures and REST api to support these resources in topo-info, topo-summary, topo-page-info, cluster, core, nimbus.
3. Display assigned cpu/mem at the index page, and show more details about the requested and assigned resources at each topology page.
4. Adjust the items shown on UI (delete topoId item, exclude topoId from schedulerInfo)

Snapshots of index page and two topology pages on UI:

image
image

image

### CODE REVIEW

**Comment by ptgoetz** [comment]
+1 looks good!

**Comment by kishorvpatil** [comment]
LGTM. The 1.8 travis build failure seems unrelated.
+1

**Comment by revans2** [comment]
The code and the UI look good.  +1 But please either file a follow on JIRA or add in documentation about the REST API changes

**Comment by zhuoliu** [comment]
Sure, @revans2 .  The REST API doc for topology summary has been updated. Thanks.

**Comment by revans2** [comment]
+1

File: STORM-UI-REST-API.md [hunk 1]
```diff
@@ -204,6 +213,14 @@ Sample response:
...
             "executorsTotal": 28,
             "replicationCount": 1
+            "requestedMemOnHeap": 640
+            "requestedMemOffHeap": 128
+            "requestedTotalMem": 768
+            "requestedCpu": 80
+            "assignedMemOnHeap": 640
+            "assignedMemOffHeap": 128
+            "assignedTotalMem": 768
+            "assignedCpu": 80
```
Comment (d2r) [inline STORM-UI-REST-API.md]: nit: need commas for valid JSON

File: storm-core/src/jvm/backtype/storm/generated/DebugOptions.java [hunk 2]
```diff
@@ -51,7 +51,7 @@
 import org.slf4j.LoggerFactory;
 
 @SuppressWarnings({"cast", "rawtypes", "serial", "unchecked"})
-@Generated(value = "Autogenerated by Thrift Compiler (0.9.2)", date = "2015-10-9")
+@Generated(value = "Autogenerated by Thrift Compiler (0.9.2)", date = "2015-10-29")
```
Comment (d2r) [inline storm-core/src/jvm/backtype/storm/generated/DebugOptions.java]: Can we eliminate diffs due only to timestamp changes?

File: storm-core/src/storm.thrift [hunk 3]
```diff
@@ -146,6 +146,12 @@ struct TopologySummary {
 513: optional string sched_status;
 514: optional string owner;
 515: optional i32 replication_count;
+521: optional double requested_memonheap;
```
Comment (d2r) [inline storm-core/src/storm.thrift]: here and elsewhere: Can we make these arguments sequential, or is there a reason we did not want to do that?

Comment (zhuoliu) [inline storm-core/src/storm.thrift]: We leave some fields for future topology-specific schedulers, etc.

File: storm-core/src/jvm/backtype/storm/scheduler/resource/ResourceAwareScheduler.java [hunk 4]
```diff
@@ -76,20 +84,30 @@ public void schedule(Topologies topologies, Cluster cluster) {
...
                         }
-                        LOG.debug("Topology: {} assigned to {} nodes on {} workers", td.getId(), nodesUsed.size(), schedulerAssignmentMap.keySet().size());
-                        cluster.setStatus(td.getId(), td.getId() + " Fully Scheduled");
+                        LOG.debug("Topology: {} assigned to {} nodes on {} workers", td.getId(), nodesUsed.size(), assignedWorkers);
+                        cluster.setStatus(td.getId(), "Fully Scheduled");
                     } catch (IllegalStateException ex) {
                         LOG.error(ex.toString());
-                        LOG.error("Unsuccessfull in scheduling {}", td.getId());
-                        cluster.setStatus(td.getId(), "Unsuccessfull in scheduling " + td.getId());
+                        LOG.error("Unsuccessful in scheduling", td.getId());
```
Comment (d2r) [inline storm-core/src/jvm/backtype/storm/scheduler/resource/ResourceAwareScheduler.java]: Thanks for fixing the typo, too.

**Comment by d2r** [comment]
+1

**Pull request merged**